analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs

```diff
@@ -445,6 +445,17 @@ public void IsTestProject_WithConfigFile(ProjectType projectType, bool expectedR
             SonarAnalysisContext.IsTestProject(context).Should().Be(expectedResult);
         }
 
+        [TestMethod]
+        public void IsUnchanged_NoData() // E.g. first scan or using old scanner version
+        {
+            var compiler = new SnippetCompiler("class Data { }", TestHelper.ProjectTypeReference(ProjectType.Product));
```

Comment:
This should not be needed. The methods exists to enable parametrized way of injecting test reference. For `Product`, it returns empty collection anyway
```suggestion
            var compiler = new SnippetCompiler("class Data { }");
```